src/keyboard-x11.c

```diff
@@ -380,19 +380,18 @@ void keyboard_SetLayout(Keyboard *pKeyboard, gint nLayout)
     }
     else
     {
-        // TODO
         gchar *sCommand;
         gchar *sId = g_slist_nth_data(pKeyboard->pPrivate->lLayoutRec, nLayout);
         gchar **lParamas = g_strsplit(sId, "+", -1);
         guint nParams = g_strv_length(lParamas);
 
         if (nParams == 1)
         {
-            sCommand = g_strdup_printf("setxkbmap -layout %s", lParamas[0]);
+            sCommand = g_strdup_printf("setxkbmap -display :0 -layout %s", lParamas[0]);
```

Comment:
Hardcoding :0 as display number will work for 95% of all cases, but LightDM can also fireup a DISPLAY :1. Also on multihead systems :0 will not always work.